dialects/autodiff/auxiliary/autodiff_rewrite_toplevel.cpp

```diff
@@ -10,7 +10,8 @@ const Def* AutoDiffEval::derive_(const Def* def) {
     auto lam    = def->as_nom<Lam>(); // TODO check if nominal
     world.DLOG("Derive lambda: {}", def);
     auto deriv_ty = autodiff_type_fun_pi(lam->type());
-    auto deriv    = world.nom_lam(deriv_ty, world.dbg(lam->name() + "_deriv"));
+    auto deriv    = world.nom_lam(deriv_ty)->set(lam->sym());
+    deriv->debug_suffix("_deriv");
```

Comment:
If I read the code correctly, this suffix is only added if the compilation runs in debug mode.
However, we always need the suffix.

Example:
```
x = pow(2,3)
y,y_pb = ad pow (2,3)
```
becomes
```
x = pow(2,3)
y,y_pb = pow_diff (2,3)
```

We need both versions (function and AD'ed function) as separate entities.

Reply:
ic. can you grep for `debug_suffix` and `debug_prefix` whether I did the same mistake somewhere else?

Reply:
I do not know if eta_exp has a problem with it.

In cps2ds, ds2cps, the `_cont` suffix is necessary. 
The `_cps` suffix is not strictly necessary (the original function should not be referenced) but the functions are only resolved in another step. They are also not semantically equivalent (another type scheme).
Therefore, I would always add the suffix.

Reply:
alright. the name *should* not really be part of the semantics, but we had this discussion that right now, we don't have a better mechanism, so in your case it makes sense. The use of `debug_*fix` in ssa_constr and eta_exp is intentional.